### PR TITLE
Create tcld reference

### PR DESCRIPTION
**Pull request by djmagee**
## What does this PR do?

Create ~~a working draft of~~ the tcld reference and installation instructions.

**PREVIEW** https://deploy-preview-1034--mystifying-fermi-1bc096.netlify.app/docs/cloud/tcld/

## Notes to reviewers

~~This PR does not yet include the sidebar.~~

### CODE REVIEW

File: docs/cloud/tcld/index.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+id: index
+title: What is tcld?
+description: tcld is a command-line tool that you can use to interact with Temporal Cloud.
+tags:
+  - operation-guide
+  - tcld
+---
+
+The Temporal Cloud CLI (tcld) is a command-line tool that you can use to interact with Temporal Cloud.
```
Comment (flossypurse) [inline docs/cloud/tcld/index.md:10]: How to install?

**Review by flossypurse — APPROVED** [review]
We should look at generating a single file for this - but probably satisfies needs for now.

File: docs/cloud/tcld/index.md (hunk 1 above)
Comment (djmagee) [reply docs/cloud/tcld/index.md:16]: Added.

File: docs/cloud/tcld/login.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+description: How to log in to Temporal Cloud using tcld.
+tags:
+  - reference
+  - tcld
+---
+
+The `tcld login` command logs in a user to Temporal Cloud.
+
+Follow instructions in the browser to log in to your Temporal account.
+After successful authentication, tcld creates a file in `~/.config/tcld` that contains the JSON web token (JWT) that tcld uses to authenticate requests made to the SaaS API.
```
Comment (mattkim) [inline docs/cloud/tcld/login.md:14]: just curious is it necessary to include info about the config dir here?

Comment (mattkim) [reply docs/cloud/tcld/login.md:14]: Also this implementation will probably change over time making it hard to keep this section in the docs up to date--i might consider removing this.

File: docs/cloud/tcld/namespace/update/accepted-client-ca/set.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+```bash
+tcld namespace update accepted-client-ca set --ca-certificate <encoded_certificate>
+```
+
+### `--ca-certificate-file`
+
+_Required modifier unless `--ca-certificate` is specified_
+
+Specify a path to a PEM file that contains a base64-encoded CA certificate.
```
Comment (mattkim) [inline docs/cloud/tcld/namespace/update/accepted-client-ca/set.md:73]: remove the "base46-encoded" part, it makes it more confusing; tldr the base64 encoded body is a part of the PEM file spec. Which is not to be confused with the the previous option which is a CA cert pem file that is additionally base64 encoded.

I would just say:

"Specify a path to a CA certificate PEM file"

File: docs/cloud/tcld/namespace/update/accepted-client-ca/set.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+```bash
+tcld namespace update accepted-client-ca set --resource-version <etag> --ca-certificate <encoded_certificate>
+```
+
+### `--ca-certificate`
+
+_Required modifier unless `--ca-certificate-file` is specified_
+
+Specify a base64-encoded CA certificate.
```
Comment (mattkim) [inline docs/cloud/tcld/namespace/update/accepted-client-ca/set.md:57]: I would change this to:

"Specify a base64-encoded string of a CA certificate PEM file."

File: docs/cloud/tcld/namespace/update/search-attributes/add.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+- Int
+- Keyword
+- Text
+
+Alias: `--sa`
+
+**Example**
+
+```bash
+tcld namespace update search-attributes add --search-attribute "Category=Text" --search-attribute "LengthOverall=Double"
```
Comment (mattkim) [inline docs/cloud/tcld/namespace/update/search-attributes/add.md:71]: nit: but these examples for the search attribute name confuses me (talking about "Category" and "LengthOverall" here).

Maybe something like "YourSearchAttributeName1=Text" and "YourSearchAttributeName2=Double"?

**Review by mattkim — APPROVED** [review]
Approved with some comments.

Additionally is it too late to add a section for "Performing a Certificate Rollover"?

Customer's have/and are going to ask us alot about this.

Here is my pending PR with instructions:

https://github.com/temporalio/tcld/pull/15

File: docs/cloud/tcld/login.md (hunk 2 above)
Comment (djmagee) [reply docs/cloud/tcld/login.md:14]: I'm good with removing it.

File: docs/cloud/tcld/namespace/update/accepted-client-ca/set.md (hunk 3 above)
Comment (djmagee) [reply docs/cloud/tcld/namespace/update/accepted-client-ca/set.md:73]: Great catch; thanks!

**Comment by djmagee** [comment]
Thanks for the review, @mattkim! I was going to ask whether you wanted to include the cert rollover process in this PR, and I see you answered my question before I asked.

**Pull request merged**